cpp/memilio/utils/date.h

```diff
@@ -105,6 +106,18 @@ struct Date {
     }
     //@}
 
+    /**
+     * Formats the date into a string in ISO format (YYYY-MM-DD).
+     * @return A string representing the date in ISO format.
+     */
+    std::string to_iso_string() const
+    {
+        std::ostringstream oss;
+        oss << std::setw(4) << std::setfill('0') << year << '-' << std::setw(2) << std::setfill('0') << month << '-'
+            << std::setw(2) << std::setfill('0') << day;
+        return oss.str();
```

Comment:
```suggestion
        // the format after ":" reads as 1) fill with "0", 2) align text right ">", 3) use a width of "4" (or "2")
        return fmt::format("{:0>4}-{:0>2}-{:0>2}", year, month, day);
```
I personally prefer format strings over streams, but both work fine. Often fmt is a lot faster than stringstream, but I have neither tested it nor does it matter that much for output.

You can choose the version you prefer.